test/formatting_stack/test_helpers.clj

```diff
@@ -8,3 +9,10 @@
   (str "file:" (-> filename
                    File.
                    .getAbsolutePath)))
+
+(defn with-mocked-diff-path
+  "Fixture to stub the absolute path in #'util.diff/unified-diff"
+  [t]
+  (with-redefs [util.diff/to-absolute-path (fn [filename]
```

Comment:
Would be nice to keep the codebase `with-redefs`-free

Maybe `unified-diff` could look like this:

```clj
(def *to-absolute-path-fn* to-absolute-path)

(speced/defn ^string? unified-diff
  "derives a patch derived from the original and revised file contents in a Unified Diff format"
  ([^string? filename ^string? original ^string? revised]
   (letfn [(lines [s]
             (string/split s #"\n"))
           (unlines [ss]
             (string/join "\n" ss))]
     (unlines (DiffUtils/generateUnifiedDiff
               (->> filename *to-absolute-path-fn* (str "a"))
               (->> filename *to-absolute-path-fn* (str "b"))
...
```